### ISSUE TITLE
Undocumented `build` command does not perform dependency resolution.

### ISSUE THREAD

**Issue by Ninetainedo** [Microsoft/vcpkg#73]
`.\vcpkg.exe build libpng` fails.

From the logfile:
> Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)

From the console log of the command, vcpkg doesn't try to install zlib before configuring libpng.

**Comment by Ninetainedo** [comment]
Same issue with libwebsockets. This is strange but it won't download the dependencies.

**Comment by ras0219-msft** [comment]
This is an issue with the (currently undocumented) build command. `vcpkg install` handles dependency resolution before delegating to the `build` command.

**Comment by alexkaratarakis** [comment]
Dependencies must be installed to get picked up during dependency resolution, and `build` does not install packages automatically, whereas `install` does

Currently the behavior is:
`vcpkg install libpng` will result in: zlib==installed and libpng==installed. 
`vcpkg build libpng` will fail if zlib!=installed
`vcpkg build libpng` will result in zlib==installed (if it already was) and libpng==built-but-not-installed

This can also be achieved unconditionally via:
`vcpkg install libpng`
`vcpkg remove libpng`

Do you think it would be intuitive to have `vcpkg build package-x` install all dependencies of package-x and then build package-x without installing it?

**Comment by Ninetainedo** [comment]
No, I definitely agree that this behavior seems to be the proper one.
But without any kind of error message, I was kinda surprised, my bad.

**Comment by alexkaratarakis** [comment]
However, you are right that the error message used to be unhelpful. Fixed after:  5b89712
Modified the command so that it doesn't start building if the dependencies are not already installed, as opposed to the previous behavior where it would fail during the build. It will now display something like this:
```
C:\path\to\vcpkg>vcpkg build opencv
The build command requires all dependencies to be already installed.
The following dependencies are missing:

    zlib
    libpng
    libjpeg-turbo
```

Thanks for reporting!

**Comment by alexkaratarakis** [comment]
Changed the output to show the arch along with the missing package name, i.e:

```
C:\path\to\vcpkg> vcpkg build opencv
The build command requires all dependencies to be already installed.
The following dependencies are missing:

    libjpeg-turbo:x86-windows
```
